### PR TITLE
[IMP] website_sale_loyalty: Convert all jQuery code to use Vanilla JS

### PR DESCRIPTION
**Pull request by visp-odoo**
This PR aim to convert all jQuery code into Vanilla JS in website_sale_loyalty, this way we will reduce the dependency of jQuery in Odoo codebase.

task-3770362

### CODE REVIEW

File: addons/website_sale_loyalty/static/src/js/coupon_toaster_widget.js [hunk 1]
```diff
@@ -11,24 +11,24 @@ const CouponToasterWidget = publicWidget.Widget.extend({
...
-            if ($title.length) {
-                Object.assign(options, {title: $title[0].innerHTML});
+        if (contentEl.length) {
+            message = contentEl.innerHTML;
+            if (titleEl.length) {
+                Object.assign(options, { title: titleEl.innerHTML });
             }
-        } else if ($title.length) {
-            message = $title[0].innerHTML;
+        } else if (titleEl.length) {
```
Comment (bso-odoo) [inline addons/website_sale_loyalty/static/src/js/coupon_toaster_widget.js:23]: I do not think `contentEl.length` and `titleEl.length` return what you expect.

Comment (visp-odoo) [reply addons/website_sale_loyalty/static/src/js/coupon_toaster_widget.js:23]: Yes, you are rigth. I think it skipped my mind while converting :sweat_smile:.

Comment (visp-odoo) [reply addons/website_sale_loyalty/static/src/js/coupon_toaster_widget.js]: @bso-odoo Done changes :+1:

**Review by AntoineVDV — COMMENTED** [review]
Hello, LGTM except that there is no need to change simple quotes into double quotes. I think you can also drop the `El` suffix from variable names, but I don't have a strong opinion on that.

**Comment by visp-odoo** [comment]
Hello @AntoineVDV,

We received feedback on our previous PR regarding the use of the `El` suffix for variable names [[REF](https://github.com/odoo/odoo/pull/158033#discussion_r1572353819)] and the preference for double quotes [[REF](https://github.com/odoo/odoo/pull/158033#discussion_r1572410232)]. Accordingly, I have followed these guidelines in this PR.

**Comment by AntoineVDV** [comment]
> use of the `El` suffix for variable names [[REF](https://github.com/odoo/odoo/pull/158033#discussion_r1572353819)]

As I said I don't have a strong opinion on that so both are fine by me.

> the preference for double quotes [[REF](https://github.com/odoo/odoo/pull/158033#discussion_r1572410232)]

However, I disagree with this one. This is, in fact, not a global guideline; it's more like a team-specific convention. A golden rule is not to enforce team-specific conventions in other teams' scopes. I understand the confusion, but it's important to make the distinction between [company-wide guidelines](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html) and team-specific conventions :sweat_smile:

**Comment by bso-odoo** [comment]
> However, I disagree with this one. This is, in fact, not a global guideline; it's more like a team-specific convention. A golden rule is not to enforce team-specific conventions in other teams' scopes. I understand the confusion, but it's important to make the distinction between [company-wide guidelines](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html) and team-specific conventions 😅

Linked beside that link is <https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines> which states:
```
Deviations from previous guidelines
We no longer distinguish between technical and non-technical string with their quote types:
use double quotes for all strings (such as "Hello"), unless they would require escaping quotes,
in which case you may use the single quotes (eg: 'He said "I love you"') or a template string
(`don't say "yes" too quickly`). This is enforced by the eslint config.
```

**Comment by AntoineVDV** [comment]
> Linked beside that link is https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines which states:
> 
> ```
> Deviations from previous guidelines
> We no longer distinguish between technical and non-technical string with their quote types:
> use double quotes for all strings (such as "Hello"), unless they would require escaping quotes,
> in which case you may use the single quotes (eg: 'He said "I love you"') or a template string
> (`don't say "yes" too quickly`). This is enforced by the eslint config.
> ```

I'm quite surprised to find they updated this page to include arbitrary guidelines, knowing that it's referenced in the documentation. In my opinion, it was not the place to recommend team-specific guidelines. The future version of the global guideline will simply omit them.

**Comment by msh-odoo** [comment]
@AntoineVDV @bso-odoo We mostly did same changes in all Remove jQuery related tasks, converted all single quotes to double quotes and suffixed the variables with ```El``` or ```Els```, can you conclude us what is the standard guidelines so we can do changes according to that.

BTW, I was checking the Web Framework code and we mostly used double quotes there so it seems that we have make double quote standard.

**Comment by bso-odoo** [comment]
@msh-odoo Let's restore/keep the quotes as they were for the changes in the `*sale*` modules.

Maybe someday there will be a few more entries in manifest files to know about specific rules and avoid such iterations.
```
    'guidelines': some url,
    'linters': {
        'js': some url,
        'py': some url,
        ...
    },
```

**Comment by visp-odoo** [comment]
@AntoineVDV I have restored the quotes changes to single quote.

**Comment by AntoineVDV** [comment]
Thanks :) Sorry for raising the question of quote styling here; I didn't expect it to turn into a "debate".

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by bso-odoo** [comment]
@robodoo r+

**Pull request closed without merging**